packages/launchpad/src/test/launchpad.tsx

```diff
@@ -12,38 +16,57 @@ jest.mock( '@automattic/data-stores', () => {
 				isFetchedAfterMount: true,
 				data: {
 					checklist: [
-						buildTask( { id: 'task1' } ),
-						buildTask( { id: 'task2' } ),
-						buildTask( { id: 'task3' } ),
+						buildTask( { id: 'task1', title: 'Task 1', completed: false, disabled: false } ),
+						buildTask( { id: 'task2', title: 'Task 2' } ),
+						buildTask( { id: 'task3', title: 'Task 3' } ),
 					],
 				},
 			};
 		} ),
 	};
 } );
 
-describe( 'Launchpad', () => {
-	describe( 'when no taskFilter is provided', () => {
-		it( 'then all tasks from useLaunchpad are rendered', () => {
-			render( <LaunchpadInternal siteSlug="any site" /> );
-			const checklistItems = screen.queryAllByRole( 'listitem' );
-			expect( checklistItems.length ).toBe( 3 );
-		} );
+describe( 'LaunchpadInternal', () => {
+	const defaultProps = {
+		siteSlug: 'any site',
+		checklistSlug: 'some-site',
+		launchpadContext: 'some-context',
+	};
+	const renderComponent = ( props: Partial< LaunchpadInternalProps > = {} ) => {
+		return render( <LaunchpadInternal { ...defaultProps } { ...props } /> );
+	};
+
+	it( 'renders the tasks', () => {
+		renderComponent();
+		const checklistItems = screen.queryAllByRole( 'listitem' );
+		expect( checklistItems.length ).toBe( 3 );
 	} );
 
-	describe( 'when taskFilter is provided', () => {
-		it( 'then render only tasks returned from the callback', () => {
-			const filter = ( tasks: Task[] ) => {
-				// return only the first task
-				return [ tasks[ 0 ] ];
-			};
+	it( 'records the checklist item click when autoTracking is enabled', async () => {
+		renderComponent( { enableAutoTracking: true } );
 
-			render( <LaunchpadInternal siteSlug="any site" taskFilter={ filter } /> );
-			const checklistItems = screen.queryAllByRole( 'listitem' );
-			expect( checklistItems.length ).toBe( 1 );
+		jest.clearAllMocks();
+		await userEvent.click( screen.getByRole( 'button', { name: /Task 1/ } ) );
 
-			const taskId = checklistItems[ 0 ].querySelector( 'button' )?.getAttribute( 'data-task' );
-			expect( taskId ).toBe( 'task1' );
+		await waitFor( () => {
+			expect( recordTracksEvent ).toHaveBeenCalledWith(
+				'calypso_launchpad_task_clicked',
+				expect.any( Object )
```

Comment:
Do we want to validate that it's supplying the correct values?

Reply:
Fixed here 8a5f37a2ca87bd188a51d2e2d7aee1924201ba2c